assets/src/components/messages/CreateAccountPopup.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import { CSSTransition } from 'react-transition-group';
+
+export interface CreateAccountPopupProps {
+  sectionSlug?: string;
+}
+
+export function CreateAccountPopup(props: CreateAccountPopupProps): JSX.Element {
+  const [show, setShow] = useState(false);
+  const [firstRender, setFirstRender] = useState(true);
+
+  const onDismiss = () => {
+    setShow(false);
+    sessionStorage.setItem('createAccountPrompt', 'hide');
+  }
+
+  const onDontAskAgain = () => {
+    setShow(false);
+    localStorage.setItem('createAccountPrompt', 'hide');
+  };
+
+  const onBlurClick = (e: any) => {
+    if (!e.createAccountPopupClick) {
+      setShow(false);
+      window.removeEventListener('click', onBlurClick);
+    }
+  }
+
+  useEffect(() => {
+    if (firstRender) {
+      const showPref = sessionStorage.getItem('createAccountPrompt') != 'hide'
+        && localStorage.getItem('createAccountPrompt') != 'hide'
+      setShow(showPref);
+      window.addEventListener('click', onBlurClick);
+    }
+    setFirstRender(false);
+  });
```

Comment:
I only recently learned of this myself, but you can give `useEffect` an empty array as a second argument which causes the effect to run only one time after the initial render. So the pattern of introducing a boolean state variable is necessary to get the "run just once" semantics.

Reply:
See here: 

https://stackoverflow.com/questions/53120972/how-to-call-loading-function-with-react-useeffect-only-once

Reply:
Great tip, thanks!